### PR TITLE
add energy loop module and reorganize some stuff

### PR DESCRIPTION
**Pull request by trevorm4**
### Description

Adding module for energy loop talent

### Testing

image
image

### CODE REVIEW

File: src/analysis/retail/evoker/preservation/modules/talents/EssenceBurst.tsx [hunk 1]
```diff
@@ -11,17 +11,27 @@ import { SPELL_COLORS } from 'analysis/retail/evoker/preservation/constants';
...
 import { t } from '@lingui/macro';
+import ItemManaGained from 'parser/ui/ItemManaGained';
 
 const ESSENCE_COSTS: { [name: string]: number } = {
   'Emerald Blossom': 3,
   Echo: 2,
   Disintegrate: 3,
 };
 
+const MANA_COSTS: { [name: string]: number } = {
```
Comment (vohrr) [inline src/analysis/retail/evoker/preservation/modules/talents/EssenceBurst.tsx:22]: The mana costs of these abilities should be a property on their SPELLS declaration

File: src/analysis/retail/evoker/preservation/modules/talents/EssenceBurst.tsx [hunk 2]
```diff
@@ -11,17 +11,27 @@ import { SPELL_COLORS } from 'analysis/retail/evoker/preservation/constants';
...
+  'Emerald Blossom': 0.048,
+  Echo: 0.017,
+  Disintegrate: 0,
+};
+
+const TOTAL_MANA = 250000;
```
Comment (vohrr) [inline src/analysis/retail/evoker/preservation/modules/talents/EssenceBurst.tsx]: Verify that the mana returns are based off Base Mana and not Enchanted mana

**Review by vohrr — APPROVED** [review]
Couple of nits, otherwise LGTM

File: src/analysis/retail/evoker/preservation/modules/talents/EssenceBurst.tsx (hunk 2 above)
Comment (trevorm4) [reply src/analysis/retail/evoker/preservation/modules/talents/EssenceBurst.tsx]: Was just using this to caclulate manaCost, shouldn't be necessary when I switch to using spellbook mana cost.

**Pull request merged**